migrations/postgres/20240901074155_trigger-dests.sql

```diff
@@ -0,0 +1,41 @@
+-- +goose Up
+-- modify "triggers" table
+ALTER TABLE "triggers" ALTER COLUMN "connection_id" DROP NOT NULL, ADD COLUMN "source_type" text NULL, ADD COLUMN "webhook_slug" text NULL;
+-- create index "idx_triggers_source_type" to table: "triggers"
+CREATE INDEX "idx_triggers_source_type" ON "triggers" ("source_type");
+-- create index "idx_triggers_webhook_slug" to table: "triggers"
+CREATE UNIQUE INDEX "idx_triggers_webhook_slug" ON "triggers" ("webhook_slug");
+-- modify "events" table
+ALTER TABLE "events" ADD COLUMN "destination_id" uuid NOT NULL, ADD COLUMN "trigger_id" uuid NULL, ADD
+ CONSTRAINT "fk_events_trigger" FOREIGN KEY ("trigger_id") REFERENCES "triggers" ("trigger_id") ON UPDATE NO ACTION ON DELETE NO ACTION;
+-- create index "idx_events_destination_id" to table: "events"
+CREATE INDEX "idx_events_destination_id" ON "events" ("destination_id");
+-- create index "idx_events_trigger_id" to table: "events"
+CREATE INDEX "idx_events_trigger_id" ON "events" ("trigger_id");
+-- drop index "idx_connection_id_event_type" from table: "signals"
+DROP INDEX "idx_connection_id_event_type";
+-- modify "signals" table
+ALTER TABLE "signals" ALTER COLUMN "signal_id" TYPE uuid, ALTER COLUMN "connection_id" DROP NOT NULL, ADD COLUMN "destination_id" uuid NOT NULL, ADD COLUMN "trigger_id" uuid NULL, ADD
```

Comment:
this won't pass on a non empty database
you create a column with a not null constraint but all current rows will have a null value